pending/20-dancer2-plugin-minion.pod

```diff
@@ -0,0 +1,232 @@
+
+This creates a task, C<thumbnails>, to automatically generate multiple
+thumbnails when provided with an original image.
+
+    # This exposes all of the minion commands
+    if (@ARGV && $ARGV[0] eq 'minion') {
+        minion_app()->start;
+        exit 0;
+    }
+
+This bit of magic exposes the entire L<Minion> CLI to your app - so you can
+use the various subcommands like C<worker> and C<job>.
+
+    set views => '.';
+
+    get '/' => sub {
+        template 'upload';
+    };
+
+This sets up a simple file uploader app. It also tells L<Dancer2> to look for
+its templates in the same directory as the app.
+
+    post '/' => sub {
+        my $file = upload('file');
+        my $name = $file->basename;
+        my $target = path('public', $name);
+        $file->copy_to($target);
+        enqueue(thumbnails => [$target]);
+        redirect "/$name";
+    };
+
+Here, we consume a file upload, stash it on disk, and call the C<thumbnails>
+job to render multiple thumbnails. This might take a while, so we don't want
+to tie up the browser waiting for them to finish.
+
+    builder {
+        # mount the container app at /dashboard/
+        # note that the trailing slash is very important
+        mount '/dashboard/' => minion_app( '/foo/bar' )->start;
+        mount '/' => start;
+    };
+
+Finally, we create the L<Plack> application by mounting our L<Dancer2> 
+application to the root URL, and attaching the L<Minion> dashboard to the
+C</dashboard/> URL (the trailing C</> is required!). It also sets the 
+"Back to Site" link on the dashboard to the ficticious URL C</foo/bar>.
+
+=head2 Now, Let's Put it All Together
+
+Here's everything you need to make this example work (put all files in the 
+same directory):
+
+    # Save this as app.psgi
+    use Dancer2;
+    use Dancer2::Plugin::Minion;
+    use Plack::Builder;
+    use File::Basename 'fileparse';
+
+    add_task thumbnails => sub {
+        my ($job, $original) = @_;
+        require Image::Imlib2::Thumbnail;
+        my $thumb = Image::Imlib2::Thumbnail->new;
+        my ($base, $dir, $ext) = fileparse( $original, qr/\.[^.]*?$/ );
+        $_->{name} = "$base-$_->{name}" for @{ $thumb->sizes };
+        my @generated = $thumb->generate($original, $dir);
+        $job->finish(\@generated);
+    };
+
+    # This exposes all of the minion commands
+    if (@ARGV && $ARGV[0] eq 'minion') {
+        minion_app()->start;
+        exit 0;
+    }
+
+    set views => '.';
+
+    get '/' => sub {
+        template 'upload';
+    };
+
+    post '/' => sub {
+        my $file = upload('file');
+        my $name = $file->basename;
+        my $target = path('public', $name);
+        $file->copy_to($target);
+        enqueue(thumbnails => [$target]);
+        redirect "/$name";
+    };
+
+    builder {
+        # mount the container app at /dashboard/
+        # note that the trailing slash is very important
+        mount '/dashboard/' => minion_app( 'http://localhost:5000/upload' )->start;
```

Comment:
and this absolute url, while probably correct for the demo isn't really what users are going to think of as a url, I'd use example.com or northpole.org or something

Reply:
Done :)